go/libraries/doltcore/sqle/tables.go

```diff
@@ -482,6 +505,43 @@ func (p doltTablePartition) Key() []byte {
 	return []byte(strconv.FormatUint(p.start, 10) + " >= i < " + strconv.FormatUint(p.end, 10))
 }
 
+// IteratorForPartition returns a types.MapIterator implementation which will iterate through the values
+// for index = start; index < end.  This iterator is not thread safe and should only be used from a single go routine
+// unless paired with a mutex
+func (p doltTablePartition) IteratorForPartition(ctx context.Context, m types.Map) (types.MapIterator, error) {
+	return newPartitionIter(ctx, m, p.start, p.end)
+}
+
+type partitionIter struct {
+	pos  uint64
+	end  uint64
+	iter types.MapIterator
+}
+
+func newPartitionIter(ctx context.Context, m types.Map, start, end uint64) (*partitionIter, error) {
+	iter, err := m.IteratorAt(ctx, start)
```

Comment:
From what I can tell, all of the perf wins are coming here.

Reply:
It really is dependent on the test query you are running with.  Projected tables have big benefits for some queries. Some schemas perform much better with the new iterators.  There are multiple wins in this PR.